ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1136,21 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    if (!ssl_verify_client_session_version(s)) {
```

Comment:
Oh, I see. I think it's best to avoid calling `ssl_get_new_session` then. It doesn't actually do anything. Having `s->hit` be 1 and `s->session` blank also violates a pretty important invariant.